lighthouse-core/test/test-utils.js

```diff
@@ -138,9 +158,75 @@ function makeParamsOptional(fn) {
   return /** @type {(...args: RecursivePartial<TParams>) => TReturn} */ (fn);
 }
 
+/**
+ * Transparently augments the promise with inspectable functions to query its state.
+ *
+ * @template T
+ * @param {Promise<T>} promise
+ */
+function makePromiseInspectable(promise) {
+  let isResolved = false;
+  let isRejected = false;
+  /** @type {T=} */
+  let resolvedValue = undefined;
+  /** @type {any=} */
+  let rejectionError = undefined;
+  const inspectablePromise = promise.then(value => {
+    isResolved = true;
+    resolvedValue = value;
+    return value;
+  }).catch(err => {
+    isRejected = true;
+    rejectionError = err;
+    throw err;
+  });
+
+  return Object.assign(inspectablePromise, {
+    isDone() {
+      return isResolved || isRejected;
+    },
+    isResolved() {
+      return isResolved;
+    },
+    isRejected() {
+      return isRejected;
+    },
+    getDebugValues() {
+      return {resolvedValue, rejectionError};
+    },
+  });
+}
+function createDecomposedPromise() {
+  /** @type {Function} */
+  let resolve;
+  /** @type {Function} */
+  let reject;
+  const promise = new Promise((r1, r2) => {
+    resolve = r1;
+    reject = r2;
+  });
+  // @ts-expect-error: Ignore 'unused' error.
+  return {promise, resolve, reject};
+}
+
+/**
+ * In some functions we have lots of promise follow ups that get queued by protocol messages.
+ * This is a convenience method to easily advance all timers and flush all the queued microtasks.
+ */
+async function flushAllTimersAndMicrotasks(ms = 1000) {
+  for (let i = 0; i < ms; i++) {
+    jest.advanceTimersByTime(1);
+    await Promise.resolve();
+  }
+}
+
 module.exports = {
   getProtoRoundTrip,
   loadSourceMapFixture,
   loadSourceMapAndUsageFixture,
   makeParamsOptional,
+  makePromiseInspectable,
+  createDecomposedPromise,
+  flushAllTimersAndMicrotasks,
+  ...mockCommands,
```

Comment:
I'm not sure I understand the benefit to having to import from multiple locations. 

> than have a large (and growing) list in one file you have to click through for the implementation anyways?

VSCode Cmd+Click already handles the re-export and goes immediately to the definition in mock-commands already so you don't even see this list if you wanted to look at the impl of `createMockSendCommandFn`. Maybe I'm misunderstanding what you're suggesting?

Reply:
> I'm not sure I understand the benefit to having to import from multiple locations.

Obviously it's also perfectly usable if you'd rather have them in there, but keeping task-specific functions together in a module and then importing them from that module when needed (instead of from one giant namespace that has all possible functions) doesn't seem all that weird to me :)